### PR TITLE
Problem: the integration tests are outdated and are failing

### PR DESCRIPTION
**Pull request by thomas-nguy**
- Update go.mod to use latest version of the repository
- Fix integration tests after dependency changes
- Fix docker to use the latest version of module/ orchestrator and ethereum contract
- Set github secret for Archive Node access key

### CODE REVIEW

**Comment by thomas-nguy** [comment]
converted as a draft as the integration cannot be run without proper setting

File: .github/workflows/integration-tests.yml [hunk 1]
```diff
@@ -284,3 +284,4 @@ jobs:
         run: ./integration_tests/integration_tests.test -test.failfast -test.v -test.run IntegrationTestSuite -testify.m Test${{ matrix.test_type }}
         env:
           ARCHIVE_NODE_URL: https://eth-mainnet.alchemyapi.io/v2/${{ secrets.ALCHEMY_ID }}
+          E2E_SKIP_CLEANUP: true
```
Comment (thomas-nguy) [inline .github/workflows/integration-tests.yml:287]: Some permission issue when removing the folder when cleaning up. Bypassing that way for now

Comment (tomtau) [reply .github/workflows/integration-tests.yml:287]: could not cleaning cause any issues?

File: go.mod [hunk 2]
```diff
@@ -3,15 +3,18 @@ module github.com/peggyjv/gravity-bridge
 go 1.15
```
Comment (tomtau) [inline go.mod:3]: can more recent Go be used?

File: go.mod [hunk 3]
```diff
@@ -3,15 +3,18 @@ module github.com/peggyjv/gravity-bridge
...
 
 require (
-	github.com/cosmos/cosmos-sdk v0.45.3
+	github.com/cosmos/cosmos-sdk v0.46.0
```
Comment (tomtau) [inline go.mod:6]: 0.46.3?

File: go.mod [hunk 4]
```diff
@@ -3,15 +3,18 @@ module github.com/peggyjv/gravity-bridge
...
 	github.com/cosmos/go-bip39 v1.0.0
-	github.com/ethereum/go-ethereum v1.10.11
+	github.com/ethereum/go-ethereum v1.10.17
```
Comment (tomtau) [inline go.mod:8]: it has this issue: https://github.com/advisories/GHSA-rqmg-hrg4-fm69 but that would be more relevant to the orchestrator/relayer which isn't depending on go-ethereum?

File: integration_tests/ethereum/contracts/CosmosToken.sol [hunk 5]
```diff
@@ -46,14 +47,15 @@ contract CosmosERC20 is ERC20 {
...
+	 * @dev Sets the gravity contract and transfer the balance to the new address.
 	 *
 	 * Requirements:
 	 *
 	 * - `msg.sender` must be the current gravity contract
 	 */
 	function setGravityContract(address _gravityAddress) external onlyGravity {
-
+		require(balanceOf(_gravityAddress) == 0, "new gravity address balance should be zero");
+		_transfer(gravity, _gravityAddress, balanceOf(gravity));
```
Comment (tomtau) [inline integration_tests/ethereum/contracts/CosmosToken.sol]: why it needs to transfer?

**Review by tomtau — COMMENTED** [review]
The Go code changes are ok, but not sure about the Ethereum contract-related ones? that contract code is only used in integration tests, so should be ok or does it need extra dApp expert review?

**Comment by thomas-nguy** [comment]
```
The Go code changes are ok, but not sure about the Ethereum contract-related ones? that contract code is only used in integration tests, so should be ok or does it need extra dApp expert review?
```

The solidity contract is the copy paste as the one already reviewed in the solidity folder. This one is in a subdirectory of `integration_tests` and uses to build the ethereum docker container. Imo there is no need to extra reviewers

File: go.mod (hunk 3 above)
Comment (thomas-nguy) [reply go.mod:6]: I have matched the dependencies in the parent folder to be the same as the module dependencies for consistency.

We could upgrade them in a separate PR

File: integration_tests/ethereum/contracts/CosmosToken.sol (hunk 5 above)
Comment (thomas-nguy) [reply integration_tests/ethereum/contracts/CosmosToken.sol]: this is related to this issue

https://github.com/crypto-org-chain/gravity-bridge/pull/93

this method only set the owner but the managed balance are not transferred

**Comment by tomtau** [comment]
why are they copy-pasted / need to be duplicated or why can't the integration tests just use the ones in the solidity folder?

File: .github/workflows/integration-tests.yml (hunk 1 above)
Comment (thomas-nguy) [reply .github/workflows/integration-tests.yml:287]: From my understanding, GitHub-hosted runners would provide a clean instance for every job execution, so I assume the cleaning will be done automatically but I will check further

**Comment by thomas-nguy** [comment]
```
why are they copy-pasted / need to be duplicated or why can't the integration tests just use the ones in the solidity folder?
```

Yeah its maybe possible but requires some change in the project structure or in the config. Will look into it. The `integration_tests/ethereum` folder and `solidity` folder use different hardhat-config which both look at contracts folder to build the ABIs. Besides the gravity contract and cosmosToken contract, all the other contracts are specific to the folder tests.

**Review by tomtau — APPROVED** [review]

**Comment by thomas-nguy** [comment]
merging this

will update the dependencies in another PR

**Pull request merged**